lib/move-sync/__tests__/move-sync.test.js

```diff
@@ -0,0 +1,411 @@
+    }
+
+    const contents = fs.readFileSync(dest + '/another-folder/file3', 'utf8')
+    const expected = /^knuckles\r?\n$/
+    assert.ok(contents.match(expected), `${contents} match ${expected}`)
+    tearDownMockFs('EISDIR')
+  })
+
+  it('should overwrite folders across devices', () => {
+    const src = `${TEST_DIR}/a-folder`
+    const dest = `${TEST_DIR}/a-folder-dest`
+
+    fs.mkdirSync(dest)
+
+    setUpMockFs('EXDEV')
+
+    try {
+      fse.moveSync(src, dest, {overwrite: true})
+    } catch (err) {
+      assert.ifError(err)
+    }
+
+    const contents = fs.readFileSync(dest + '/another-folder/file3', 'utf8')
+    const expected = /^knuckles\r?\n$/
+    assert.ok(contents.match(expected), `${contents} match ${expected}`)
+    tearDownMockFs('EXDEV')
+  })
+
+  it('should move folders across devices with EXDEV error', () => {
+    const src = `${TEST_DIR}/a-folder`
+    const dest = `${TEST_DIR}/a-folder-dest`
+
+    setUpMockFs('EXDEV')
+
+    try {
+      fse.moveSync(src, dest)
+    } catch (err) {
+      assert.ifError(err)
+    }
+
+    const contents = fs.readFileSync(dest + '/another-folder/file3', 'utf8')
+    const expected = /^knuckles\r?\n$/
+    assert.ok(contents.match(expected), `${contents} match ${expected}`)
+    tearDownMockFs('EXDEV')
+  })
+
+  it('should move folders across devices with EPERM error', () => {
+    const src = `${TEST_DIR}/a-folder`
+    const dest = `${TEST_DIR}/a-folder-dest`
+
+    setUpMockFs('EPERM')
+
+    try {
+      fse.moveSync(src, dest)
+    } catch (err) {
+      assert.ifError(err)
+    }
+
+    const contents = fs.readFileSync(dest + '/another-folder/file3', 'utf8')
+    const expected = /^knuckles\r?\n$/
+    assert.ok(contents.match(expected), `${contents} match ${expected}`)
+    tearDownMockFs('EPERM')
+  })
+
+  it('should move folders across devices with ENOTSUP error', () => {
+    const src = `${TEST_DIR}/a-folder`
+    const dest = `${TEST_DIR}/a-folder-dest`
+
+    setUpMockFs('ENOTSUP')
+
+    try {
+      fse.moveSync(src, dest)
+    } catch (err) {
+      assert.ifError(err)
+    }
+
+    const contents = fs.readFileSync(dest + '/another-folder/file3', 'utf8')
+    const expected = /^knuckles\r?\n$/
+    assert.ok(contents.match(expected), `${contents} match ${expected}`)
+    tearDownMockFs('ENOTSUP')
+  })
+
+  describe('clobber', () => {
+    it('should be an alias for overwrite', () => {
+      const src = `${TEST_DIR}/a-file`
+      const dest = `${TEST_DIR}/a-folder/another-file`
+
+      // verify file exists already
+      assert(fs.existsSync(dest))
+
+      try {
+        fse.moveSync(src, dest, {clobber: true})
+      } catch (err) {
+        assert.ifError(err)
+      }
+
+      const contents = fs.readFileSync(dest, 'utf8')
+      const expected = /^sonic the hedgehog\r?\n$/
+      assert.ok(contents.match(expected), `${contents} match ${expected}`)
+    })
+  })
+
+  describe.skip('> when trying to a move a folder into itself', () => {
```

Comment:
Because **prevent copying into itself** is not implemented yet for `copySync`. But as I am thinking now, that would work only for the cases if we move across devices. But, when src and dest are on the same device, we still should be able to prevent **moving into itself**. Is that right?